esti/catalog_export_test.go

```diff
@@ -0,0 +1,418 @@
+package esti
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/glue"
+	"github.com/aws/aws-sdk-go-v2/service/glue/types"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/api/apigen"
+	"github.com/treeverse/lakefs/pkg/api/apiutil"
+	"github.com/treeverse/lakefs/pkg/block"
+	"github.com/treeverse/lakefs/pkg/testutil"
+	"gopkg.in/yaml.v3"
+)
+
+type schemaField struct {
+	Name       string            `yaml:"name"`
+	Type       string            `yaml:"type"`
+	IsNullable bool              `yaml:"nullable"`
+	Metadata   map[string]string `yaml:"metadata,omitempty"`
+}
+type tableSchema struct {
+	Type   string        `yaml:"type"`
+	Fields []schemaField `yaml:"fields"`
+}
+
+type hiveTableSpec struct {
+	Name             string      `yaml:"name"`
+	Type             string      `yaml:"type"`
+	Path             string      `yaml:"path"`
+	PartitionColumns []string    `yaml:"partition_columns"`
+	Schema           tableSchema `yaml:"schema"`
+}
+
+type exportHooksTestData struct {
+	SymlinkScriptPath   string
+	TableDescriptorPath string
+	GlueScriptPath      string
+	Branch              string
+	GlueDB              string
+	AccessKeyId         string
+	SecretAccessKey     string
+	OverrideCommitID    string
+	TableSpec           *hiveTableSpec
+}
+
+const glueExportScript = `
```

Comment:
Can you document this please?  Docs should say it's Lua (I think...), but also explain what `{{...}}` in it means.

Reply:
As an alternative, Go has a statically-linked readonly filesystem available.  If you used that you could put this in a file, statically link that file into the filesystem, and then the test could read from this file.

Reply:
Changed all the templates / strings everything  to files with go template!